packages/pwa/app/utils/routes-utils.js

```diff
@@ -5,38 +5,114 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-// import {getUrlConfig} from './utils'
+import {getDefaultSite, getSites} from './site-utils'
 import {urlPartPositions} from '../constants'
 
 /**
- * Configure the routes based on url configuration provided.
+ * Construct literal routes based on url config
+ * with each site id/alias and locale id/alias pair from sites config
  *
  * @param {array} routes - array of routes to be configured
- * @param {object} urlConfig - the url configuration object
- * @param {object} - a custom configured object
+ * @param {object} urlConfig
+ * @param {object} options
+ * @param {array} options.ignoredRoutes - routes that does not need configured
  * @return {array} - list of configured route objects
  */
 export const configureRoutes = (routes = [], urlConfig, {ignoredRoutes = []}) => {
```

Comment:
```suggestion
export const generateRoutePermutations = (routes = [], urlConfig, {ignoredRoutes = []}) => {
```

What if we rename the function like that? I think it'll make a clearer picture:
- That when you call the function, you will get a lot more routes than what you put in. `configure` may sound like you're doing something to the given routes, but I don't think it paints a picture that you're generating new routes.

Reply:
Let's leave it up for follow-up discussions after this get merged